stdlib/public/core/StringBridge.swift

```diff
@@ -410,19 +414,28 @@ extension String {
   }
 }
 
+@_effects(releasenone)
+private func _createNSString(
+  _ receiver: _StringSelectorHolder,
+  _ ptr: UnsafePointer<UInt8>,
+  _ count: Int,
+  _ encoding: UInt32
+) -> AnyObject? {
+  return receiver.createTaggedString(bytes: ptr, count: count)
+}
+
 @_effects(releasenone)
 private func _createCFString(
   _ ptr: UnsafePointer<UInt8>,
   _ count: Int,
   _ encoding: UInt32
-) -> AnyObject {
-  return _swift_stdlib_CFStringCreateWithBytes(
-    nil, //ignored in the shim for perf reasons
+) -> AnyObject? {
+  return _createNSString(
+    unsafeBitCast(__StringStorage.self as AnyClass, to: _StringSelectorHolder.self),
```

Comment:
Using '__StringStorage.self as AnyClass' instead of my original '_StringSelectorHolder.self as AnyObject' avoids a bunch of machinery that appeared to be boxing something.

Unfortunately it still emits a metadata accessor call that costs about 2/3 as much as objc_msgSend, but I haven't found a good way to avoid that yet.